### PR TITLE
Allow the key delimiter to be specified.

### PR DESCRIPTION
**Pull request by alex-code**
Expand `ArrayHelper::getValue` to allow the key delimiter to be specified.

``` php
$array = [
  'top' => [
    '1.0' => 'value1',
    '2.0' => 'value2'
  ]
];

ArrayHelper::getValue($array, 'top.1.0') // returns null
ArrayHelper::getValue($array, 'top|1.0', null, '|') // returns 'value1'
```

### CODE REVIEW

**Comment by samdark** [comment]
I think we should introduce escaped dot such as `\.`.

**Comment by klimov-paul** [comment]
Why don't simply use array as a key parameter?
```php
ArrayHelper::getValue($array, ['top','1.0']);
```

**Comment by Alex-Code** [comment]
I originally thought about escaping the dot but then you need another method to escape them.
What if the key comes from the database with a dot already in it.

**Comment by qiangxue** [comment]
I think we should the code as is. If you foresee that you may use dot in a key that should not be treated as deliminator, then do not use this method. For the example you described that a key is from database, then even if you introduce a different deliminator, the new one may still have problem. Introduce escaping mechanism here is an overkill, IMO.

**Comment by Alex-Code** [comment]
I've changed it based on @klimov-paul idea.
The key can be specified as an array, this avoids the need for escaping or providing a delimiter.

**Comment by samdark** [comment]
Would be great to get tests for it in `ArrayHelperTest`.

**Comment by Alex-Code** [comment]
Added a test.
```
$array = [
  'versions' => [
    '1.0' => ['option' => 'value'],
    '2.0' => ['option' => 'value'],
  ]
];

$version = '1.0';
ArrayHelper::getValue($array, ['versions', $version, 'option']);
```

File: framework/helpers/BaseArrayHelper.php [hunk 1]
```diff
@@ -163,11 +163,12 @@ public static function merge($a, $b)
      * ~~~
      *
      * @param array|object $array array or object to extract value from
-     * @param string|\Closure $key key name of the array element, or property name of the object,
+     * @param string|\Closure|array $key key name of the array element, or property name of the object,
```
Comment (samdark) [inline framework/helpers/BaseArrayHelper.php]: Need docs on how to specify it.

File: framework/helpers/BaseArrayHelper.php [hunk 2]
```diff
@@ -163,11 +163,12 @@ public static function merge($a, $b)
...
      * or an anonymous function returning the value. The anonymous function signature should be:
      * `function($array, $defaultValue)`.
      * @param mixed $default the default value to be returned if the specified array key does not exist. Not used when
      * getting value from an object.
+     * @param string $delimiter the delimiter character for the key, defaults to a dot (.)
```
Comment (samdark) [inline framework/helpers/BaseArrayHelper.php]: It's not used anywhere.

**Comment by klimov-paul** [comment]
Resolved by commit 8caaba0

**Pull request closed without merging**